File: C_Programs/prime_check.c
#include <stdio.h>

int is_prime(long long n) {
    if (n < 2) return 0;
    if (n % 2 == 0) return n == 2;
    for (long long i = 3; i * i <= n; i += 2)
        if (n % i == 0) return 0;
    return 1;
}

int main(void) {
    long long n;
    printf("Enter an integer: ");
    if (scanf("%lld", &n) != 1) {
        fprintf(stderr, "Invalid input\n");
        return 1;
    }
    printf("%lld is %s\n", n, is_prime(n) ? "prime" : "not prime");
    return 0;
}

File: Java_Programs/PrimeCheck.java
import java.util.Scanner;

public class PrimeCheck {
    public static boolean isPrime(long n) {
        if (n < 2) return false;
        if (n % 2 == 0) return n == 2;
        for (long i = 3; i * i <= n; i += 2) if (n % i == 0) return false;
        return true;
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter an integer: ");
        if (!sc.hasNextLong()) { System.err.println("Invalid input"); sc.close(); System.exit(1); }
        long n = sc.nextLong();
        System.out.println(n + (isPrime(n) ? " is prime" : " is not prime"));
        sc.close();
    }
}
